From 3d180f778df2180ac9bc5b6093d3e10fb1b5850f Mon Sep 17 00:00:00 2001
From: Tim Edwards <tim@opencircuitdesign.com>
Date: Wed, 3 Apr 2024 11:01:58 -0400
Subject: [PATCH 4/9] Corrected an error that had previously been corrected in
 PropertyMatch() but not corrected symmetrically between circuit1 and
 circuit2;  this left the possibility that "M=1" in one circuit vs. no "M"
 entry in the other would still pop up as a property error, depending on which
 circuit (layout or schematic) was listed first.

---
 VERSION       | 2 +-
 base/netcmp.c | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/VERSION b/VERSION
index ccfe09f..a851fdf 100644
--- a/VERSION
+++ b/VERSION
@@ -1 +1 @@
-1.5.272
+1.5.273
diff --git a/base/netcmp.c b/base/netcmp.c
index 9772bfc..e01a065 100644
--- a/base/netcmp.c
+++ b/base/netcmp.c
@@ -6211,7 +6211,9 @@ PropertyMatch(struct Element *E1, struct Element *E2,
 	        if (kl1 != NULL)
 		    break;	// Property is required
 	    }
-	    else if (vl1->value.ival != 1)
+	    else if ((vl1->type == PROP_INTEGER) && (vl1->value.ival != 1))
+		break;	// Property M != 1 or S != 1 is a mismatch.
+	    else if ((vl1->type == PROP_DOUBLE) && (vl1->value.dval != 1))
 		break;	// Property M != 1 or S != 1 is a mismatch.
 	 }
 	 if (vl1->type != PROP_ENDLIST) {

